components/payments/cmd/connectors/internal/connectors/modulr/task_fetch_beneficiaries.go

```diff
@@ -15,13 +15,30 @@ import (
 	"github.com/formancehq/payments/cmd/connectors/internal/task"
 )
 
+type fetchBeneficiariesState struct {
+	LastCreated string `json:"last_created"`
+}
+
+func (s *fetchBeneficiariesState) SetLatest(latest *client.Beneficiary) error {
+	lastCreatedTime, err := time.Parse("2006-01-02T15:04:05.999-0700", latest.Created)
+	if err != nil {
+		return err
+	}
+	creationDate := lastCreatedTime.Format("2006-01-02T15:04:05-0700")
+	if s.LastCreated == "" || creationDate > s.LastCreated {
```

Comment:
I don't think comparing dates as strings here is a good idea, because of the timezone ("-0700") than can lead to same dates not being equal.

Reply:
Fixed, they're now stored and compared as time.Time